extensions/binding/org.eclipse.smarthome.binding.tradfri/src/main/java/org/eclipse/smarthome/binding/tradfri/internal/discovery/TradfriDiscoveryParticipant.java

```diff
@@ -61,25 +61,30 @@ public ThingUID getThingUID(ServiceInfo service) {
     @Override
     public DiscoveryResult createResult(ServiceInfo service) {
 
-        if (service.getHostAddresses() != null && service.getHostAddresses().length > 0
-                && !service.getHostAddresses()[0].isEmpty()) {
-            String ip = service.getHostAddresses()[0];
+        if (service.getType().equals(getServiceType())) {
+            if (service.getHostAddresses() != null && service.getHostAddresses().length > 0
+                    && !service.getHostAddresses()[0].isEmpty()) {
+                String ip = service.getHostAddresses()[0];
 
-            ThingUID thingUID = getThingUID(service);
-            if (thingUID != null) {
-                logger.debug("Discovered Tradfri gateway: {}", service);
-                Map<String, Object> properties = new HashMap<>(4);
-                properties.put(Thing.PROPERTY_VENDOR, "IKEA of Sweden");
-                properties.put(GatewayConfig.HOST, ip);
-                properties.put(GatewayConfig.PORT, service.getPort());
-                String fwVersion = service.getPropertyString("version");
-                if (fwVersion != null) {
-                    properties.put(Thing.PROPERTY_FIRMWARE_VERSION, fwVersion);
+                ThingUID thingUID = getThingUID(service);
+                if (thingUID != null) {
+                    logger.debug("Discovered Tradfri gateway: {}", service);
+                    Map<String, Object> properties = new HashMap<>(4);
+                    properties.put(Thing.PROPERTY_VENDOR, "IKEA of Sweden");
+                    properties.put(GatewayConfig.HOST, ip);
+                    properties.put(GatewayConfig.PORT, service.getPort());
+                    String fwVersion = service.getPropertyString("version");
+                    if (fwVersion != null) {
+                        properties.put(Thing.PROPERTY_FIRMWARE_VERSION, fwVersion);
+                    }
+                    return DiscoveryResultBuilder.create(thingUID).withProperties(properties).withLabel("TRÅDFRI Gateway")
+                            .withRepresentationProperty(GatewayConfig.HOST).build();
                 }
-                return DiscoveryResultBuilder.create(thingUID).withProperties(properties).withLabel("TRÅDFRI Gateway")
-                        .withRepresentationProperty(GatewayConfig.HOST).build();
+            } else {
+                logger.warn("Discovered ServiceInfo doesn't have ip address: {}", service);
```

Comment:
Is this a mandatory requirement for mDNS annoucements? If so, it should rather be logged by jmDNS itself already. If not, we must not log a warning here.

Reply:
I don't know if it's mandatory (I suppose it's not because I've seen several devices from different vendors to not announce their IP but return: /(null):<port> instead.

This warning is to have easier track for issues like:
"Why is my gateway not discovered"

Reply:
The user won't see this message anyhow. And note that you are currently doing this logging for ANY coap service, not only for Tradfri gateways.